src/phoenix/experimental/evals/models/openai.py

```diff
@@ -302,18 +303,23 @@ async def _async_generate_with_retry(self, **kwargs: Any) -> Any:
         @self.retry
         @self._rate_limiter.alimit
         async def _completion_with_retry(**kwargs: Any) -> Any:
-            if self._model_uses_legacy_completion_api:
-                if "prompt" not in kwargs:
-                    kwargs["prompt"] = "\n\n".join(
-                        (message.get("content") or "")
-                        for message in (kwargs.pop("messages", None) or ())
-                    )
-                # OpenAI 1.0.0 API responses are pydantic objects, not dicts
-                # We must dump the model to get the dict
-                res = await self._async_client.completions.create(**kwargs)
-            else:
-                res = await self._async_client.chat.completions.create(**kwargs)
-            return res.model_dump()
+            try:
+                if self._model_uses_legacy_completion_api:
+                    if "prompt" not in kwargs:
+                        kwargs["prompt"] = "\n\n".join(
+                            (message.get("content") or "")
+                            for message in (kwargs.pop("messages", None) or ())
+                        )
+                    # OpenAI 1.0.0 API responses are pydantic objects, not dicts
+                    # We must dump the model to get the dict
```

Comment:
Are we referring to the OpenAI Python SDK version here, or the legacy completions API? I think I'm getting confused, if we're referring to the former, maybe just call it the legacy completions API? I haven't heard that referred to as OpenAI 1.0 (although maybe I missed something).

Reply:
I'm not sure which one we're talking about, I think it's probably the legacy completions API but we should double check. Maybe we should do a pass on this when we move evals out of experimental?